code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -578,15 +578,21 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
  * - ignore_temperature. Returns a cheaper form of gas calculation, useful if the temperature difference between the two gasmixes is low or nonexistant.
  */
 /datum/gas_mixture/proc/gas_pressure_calculate(datum/gas_mixture/output_air, target_pressure, ignore_temperature = FALSE)
-	if((total_moles() <= 0) || (temperature <= 0))
+	// So we dont need to iterate the gaslist multiple times.
+	var/our_moles = total_moles()
```

Comment:
this value won't change midway through this proc right?

Reply:
Yes it shouldnt.